runtime/lua/vim/shared.lua

```diff
@@ -158,6 +158,16 @@ function vim.tbl_map(func, t)
   return rettab
 end
 
+--- Reduce a table...
+---
+function vim.tbl_reduce(acc, func, t)
+  for _, v in pairs(t) do
+    func(acc, v)
```

Comment:
don't we want `acc = func(acc, v)` to be fully general? the function can return the mutated-in-place value anyway.